File: problemas/iniciante/1158/1158.py
def somaPA(a0, n):
    a0 += 1 - (abs(a0) % 2)
    an = a0 + (n - 1) * 2

    return ((a0 + an) * n)//2


N = int(input())
for _ in range(N):
    X, Y = [int(x) for x in input().strip().split(' ')]

    print(somaPA(X, Y))

File: problemas/iniciante/1158/1158.rs
use std::io;

fn somaPA(a1: i64, an: i64, n: i64) -> i64 {
    return ((a1 + an) * n) / 2;
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input);
    let mut n: i32 = input.trim().parse().unwrap();

    for _ in 0..n {
        input = String::new();
        io::stdin().read_line(&mut input);
        let parts: Vec<&str> = input.split(" ").collect();
        let mut x: i64 = parts[0].trim().parse().unwrap();
        let y: i64 = parts[1].trim().parse().unwrap();

        x += 1 - x.abs() % 2;

        println!("{}", somaPA(x, x + 2 * (y - 1), y));
    }
}
